helpers/telemetry.py

```diff
@@ -0,0 +1,169 @@
+import asyncio
+from datetime import datetime
+
+from shared.django_apps.pg_telemetry.models import SimpleMetric as PgSimpleMetric
+from shared.django_apps.ts_telemetry.models import SimpleMetric as TsSimpleMetric
+
+from database.engine import get_db_session
+from database.models.core import Commit, Owner, Repository
+
+
+def fire_and_forget(fn):
+    """
+    Decorator for an async function that will throw it in the asyncio queue and
+    return immediately. The caller does not need to await the result.
+
+    Useful for things like telemetry where you don't want to delay the actual
+    logic too long for it and it's not the worst thing if it fails.
+    """
+    if not hasattr(fire_and_forget, "background_tasks"):
+        fire_and_forget.background_tasks = set()
+
+    def wrapper(*args, **kwargs):
+        task = asyncio.create_task(fn(*args, **kwargs))
+        fire_and_forget.background_tasks.add(task)
+        task.add_done_callback(fire_and_forget.background_tasks.discard)
+
+    return wrapper
+
+
+class MetricContext:
+    """
+    Timeseries metrics can be tagged with context (repo, commit, owner) and this
+    class holds onto that context. It exposes a way to log timeseries metrics
+    with context appended.
+
+    Create it with whatever context you've got (e.g. IDs that were passed in as
+    task arguments) and it will attempt to fetch the rest before logging for the
+    first time.
+
+    `log_simple_metric()` will call `populate()` if it hasn't been called before
+    and then log a simple metric with the pass-in name/value to both Postgres
+    and Timescale. This function is synchronous/blocking.
+
+    `attempt_log_simple_metric()` is the @fire_and_forget version of
+    `log_simple_metric()`, meaning it will throw populating/logging on the
+    asyncio queue and then immediately return without needing to be awaited. Can
+    only be used in an event loop/async function.
+    """
+
+    def __init__(
+        self,
+        repo_id: int = None,
+        owner_id: int = None,
+        commit_id: int = None,
+        commit_sha: str = None,
+    ):
+        self.repo_id = repo_id
+        self.commit_id = commit_id
+        self.commit_sha = commit_sha
+        self.owner_id = owner_id
+        self.repo_slug = None
+        self.owner_slug = None
+        self.commit_slug = None
+        self.populated = False
+
+    def populate(self):
+        if self.populated:
+            return
+
+        repo = None
+        owner = None
+        commit = None
+        dbsession = get_db_session()
+
+        if self.repo_id:
+            repo = (
```

Comment:
Will this eventually be replaced with DJango ORM equivalents?
It seems that `PgSimpleMetric` and `TsSimpleMetric` are managed by Django (probably comes from `shared`) but this stuff isn't.

Not complaining or asking changes, just curious.

Reply:
Ah it's still undecided, I see
https://github.com/codecov/engineering-team/issues/547